solver/llbsolver/bridge.go

```diff
@@ -157,6 +154,9 @@ func (b *llbBridge) Solve(ctx context.Context, req frontend.SolveRequest, sid st
 	} else {
 		return &frontend.Result{}, nil
 	}
+	if req.Evaluate {
+		_, err = res.Ref.Result(ctx)
```

Comment:
`res.Ref` can be `nil` here and `f.Solve` can return a map instead of a single ref iiuc